traefik/templates/rbac/podsecuritypolicy.yaml

```diff
@@ -13,13 +13,17 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
     app.kubernetes.io/instance: {{ .Release.Name }}
 spec:
-  privileged: false
-  allowPrivilegeEscalation: false
   requiredDropCapabilities:
     - ALL
 {{- if not .Values.securityContext.runAsNonRoot }}
   allowedCapabilities:
     - NET_BIND_SERVICE
+  allowPrivilegeEscalation: true
+  privileged: true
```

Comment:
why would you run as non-root and yet set it as privileged and allow privilege escalation?

Reply:
As a general rule: I would do this for my unprivileged process being able to bind on privileged ports.
Though I'm a little rusty on the specifics of this PR. I recall the traefik container image not setting the EP setcap on traefik binary.... Does this trick still applies, then?
I know I ended up using 8080/8443 as container ports, while exposing 80/443 as hostPorts ... my allowPrivilegeEscalation is false ...

Reply:
... And now I get it. "if not .Values.xxx.runAsNonRoot".
runAsNonRoot not set, so we run as root. So we allow for this in PSP.

Though I wouldn't recommend it, personally.
The traefik container image runs as root, by default

```
$ docker run --entrypoint /bin/sh -it docker.io/traefik
/ # id
uid=0(root) gid=0(root) groups=0(root),1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)
```

Which isn't really a good practice .. but "fixing" the image would most definitely break lots of existing installs ... I would avoid changing that bit now, to avoid the "I can't read/renew/... my certificates" cases

Thus, by default, with PSP enabled: traefik pods won't start. Either set runAsNonRoot in values, or allow for privileged containers in PSP.

Reply:
now .... in retrospective ... that should have stayed in #337 ...